pkg/server/source/github.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2025 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"crypto"
+	"errors"
+	"fmt"
+	"net/http"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	"github.com/google/go-github/v64/github"
+	"github.com/sethvargo/go-gcpkms/pkg/gcpkms"
+
+	"github.com/abcxyz/pkg/githubauth"
+)
+
+type GitHubAppConfig struct {
+	AppID      string
+	PrivateKey string
+	KMSKeyID   string
+	Signer     crypto.Signer
+}
+
+type gitHubSourceSystem struct {
+	apps []*githubauth.App
+}
+
+func newKMSSigner(ctx context.Context, keyID string) (crypto.Signer, error) {
+	client, err := kms.NewKeyManagementClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup client: %w", err)
+	}
+	signer, err := gcpkms.NewSigner(ctx, client, keyID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create signer: %w", err)
+	}
+	return signer, nil
+}
```

Comment:
I'm still not convinced we want this directly in here; we should really only care about accepting a `crypto.Signer` interface.

Reply:
I moved it up a level to the runner.